### ISSUE TITLE
UNHANDLED_REJECTION

### ISSUE THREAD

**Issue by JulesMoorhouse** [apple/sample-cloudkit-sharing#15]
I’ve tried sharing a contact however I’m getting an iCloud has stop responding error.

REPORTED ERROR TITLE
Error: ShareRecord: Unable to find applicationIdentifier for containerId = iCloud.com.mydomain.samples.cloudkit.sharing and fileExtension = undefined

That’s my main issue.

However, when I tried to share my contact with the messages app, no to address field was shown. I had to use discord.

Any advice appreciated 

Jules.

**Comment by JulesMoorhouse** [comment]
Hmm looks like I need to setup a fallback url.

Would be nice to see this in the readme.

https://stackoverflow.com/questions/50937353/cloudkit-ckshare-url-goes-nowhere

**Comment by JulesMoorhouse** [comment]
I haven’t got back to this yet, but it does sound unusual. 

However, I was thinking that some JavaScript on a GitHub pages site might work 🤷‍♂️

**Comment by fsjack** [comment]
> I haven’t got back to this yet, but it does sound unusual.
> 
> However, I was thinking that some JavaScript on a GitHub pages site might work 🤷‍♂️

yeah, Github page might work, but still it's not the way should be.

I found that when sharing with message, when the share link appears as a cloud button meaning system will take over and redirect to your app, but when you receive a usual link, system will pop up a web view, in which way error shows up.

**Comment by JulesMoorhouse** [comment]
I've you try this tutorial, there's a download 'final' with all the code, there's also some cloud kit set...
This seems to work..

https://www.raywenderlich.com/29934862-sharing-core-data-with-cloudkit-in-swiftui

Let me know how you get on.

**Comment by fsjack** [comment]
> I've you try this tutorial, there's a download 'final' with all the code, there's also some cloud kit set... This seems to work..
> 
> https://www.raywenderlich.com/29934862-sharing-core-data-with-cloudkit-in-swiftui
> 
> Let me know how you get on. I haven't looked at the code in detail, so I'm not sure what it does different.

I found a way to bypass this issue.
As I previously said, when you receive the invitation message showed as a link, it will pop up a webview and error will be presented. However if it's the cloud button it will took over by system and do the jump. The key decide whether it's cloud button or link is whether he/she the people you sent link to is your contacts or not. Easy to test, you could just tap on the name on top of the message conversation UI, see the link section on the bottom? just tap the link, it will do the jump if your application setting right. 
So just add the people you invited to your contact will solve that problem.

**Comment by JulesMoorhouse** [comment]
I’m not sure I understand.

Are you saying, analyse the link (I’m assumed you have access to this is the share controller) then create a new message ?

Confused

**Comment by JulesMoorhouse** [comment]
Are you saying, this is a user training issue?

Also you can still copy the link in share options.

I wonder if this will occur in a live app, what do you think ?

Edit: I deleted my last reply, I think I see what you mean now.
Edit2: the link I posted doesn’t fix the issue, but you you do get a To field.

**Comment by fsjack** [comment]
> Are you saying, this is a user training issue?
> 
> Also you can still copy the link in share options.
> 
> I wonder if this will occur in a live app, what do you think ?
> 
> Edit: I deleted my last reply, I think I see what you mean now. Edit2: the link I posted doesn’t fix the issue, but you you do get a To field.

Good question, I never success in other app except Message, I don't think it will work.

**Comment by JulesMoorhouse** [comment]
Ok, I’ve published the Ray Wenderlich example code to the App Store.

I started writing a post here saying this isn’t a problem, I could always get a share with icon and couldn’t see a plain link.

However, once I had shared the link and opened it on someone else’s device, it fails to establish the share in that instance of the app. It does show in the master app, that’s there’s a invite. 

Also another problem, on a new record, when I open the share feature, it opens as a blank form, you have to close it and re-open it.

🤷‍♂️

**Comment by JulesMoorhouse** [comment]
@sjrmanning thanks for your reply.
I’m not at my Mac atm.
However, I haven’t changed the plist in the sample code.

Can you confirm what you mean by association?

**Comment by rindtdev** [comment]
Running into the same issue right now.

@sjrmanning I can't get raywenderlichs, this sample code or my own project running. I have named them like its recommended with com.mycompany.myapp and the container iCloud.com.mycompany.myapp they are associated on developer.apple.com and CKSharingSupported is set. I even regenerated the container, removed association on developer.apple.com, profiles etc in a lot of different steps and orders and it isn't budging - all the things I found on the web

This is the full error I get on iCloud.com site with the iCloud share link - I mean maybe Apple is currently working on stuff with that Hotfix build - but its very inconvenient because I need this... working?


```
APP NAME
cloudos2

CLIENT TIME
Wed Aug 17 2022 22:03:32 GMT+0200 (Mitteleuropäische Sommerzeit) (1660766612531)

USER AGENT
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.0 Safari/605.1.15

HOSTNAME
www.icloud.com

BUILD INFO
2220Hotfix13__2220Hotfix13__de-de

SESSION ID
d7cd5116-1074-437a-bce2-a2ba204ad07a

ENVIRONMENT INFO
PROD

REPORTED ERROR TITLE
Error: ShareRecord: Unable to find applicationIdentifier for containerId = iCloud.com.<mycompany>.MyTravelJournal and fileExtension = undefined

REPORTED ERROR TYPE
UNHANDLED_REJECTION

LINE NUMBER
11

STACK
r@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:146140
a@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:146187
_getApplicationIdentifier@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:2146666
_extractCloudKitResponse@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:2144591
init@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:2143329
create@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:137482
@https://www.icloud.com/system/cloudos2/2220Hotfix13/de-de/main.js:11:2149291
promiseReactionJob@[native code]

ERROR CONTEXT
undefined
```
